ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/QueryStreamHandler.java

```diff
@@ -82,6 +83,14 @@ public void handle(final RoutingContext routingContext) {
     final Optional<Boolean> internalRequest = ServerVerticle.isInternalRequest(routingContext);
     final MetricsCallbackHolder metricsCallbackHolder = new MetricsCallbackHolder();
     final long startTimeNanos = Time.SYSTEM.nanoseconds();
+
+    // For http2, if we get a goAway, meaning that the connection is going to be shutdown, just end
+    // it immediately, otherwise there will be a delay of 30s until we get a call on the endhandler.
+    // Consider merging this logic into ConnectionQueryManager used by push queries.
```

Comment:
It might be nice to track this with a ticket, if I'm reading it correctly as a TODO.